### PR TITLE
tickets/DM-44689: Add resource limits/requests to prepulled pods

### PR DESCRIPTION
**Pull request by athornton**
Add limits and requests to prepulled pods.

### CODE REVIEW

**Comment by athornton** [comment]
Rebase when DM-44654 merged.  Also do scriv collect at that point.

File: controller/src/controller/services/builder/prepuller.py [hunk 1]
```diff
@@ -65,6 +66,16 @@ def build_pod(self, image: RSPImage, node: str) -> V1Pod:
...
+                        resources=V1ResourceRequirements(
+                            limits={
+                                "cpu": "1",
+                                "memory": "512Mi",
+                            },
+                            requests={
+                                "cpu": "1m",
+                                "memory": "16Mi",
+                            },
+                        ),
```
Comment (rra) [inline controller/src/controller/services/builder/prepuller.py:78]: Let's set the limits to the same as the requests. Running `/bin/true` should never take that much memory.

**Review by rra — APPROVED** [review]
Looks good except let's make the limits smaller.

File: controller/src/controller/services/builder/prepuller.py (hunk 1 above)
Comment (rra) [reply controller/src/controller/services/builder/prepuller.py:78]: Also if you remove the trailing commas, ruff should  manage to make this a lot more compact.

**Pull request merged**